src/components/landing/SliderContainer.tsx

```diff
@@ -0,0 +1,468 @@
+import styled from 'styled-components';
+import Link from 'next/link';
+import { MutableRefObject, useEffect } from 'react';
+import Lottie from 'lottie-react';
+import { step1Lottie } from 'public/assets/lottie';
+import { step2Lottie } from 'public/assets/lottie';
+import { step3Lottie } from 'public/assets/lottie';
+import { imgLandingBgFirst, imgLandingBgSecond, imgLandingBgLast } from 'public/assets/images/index';
+import { imgLandingBubble } from 'public/assets/images/index';
+import { imgLandingEar, imgLandingMic, imgLandingFist } from 'public/assets/images/index';
+import { COLOR } from 'src/styles/color';
+import { FONT_STYLES } from 'src/styles/fontStyle';
+import ImageDiv from '../common/ImageDiv';
+import { useSlideObserver } from 'src/components/landing/useSlideObserver';
+
+interface SliderContainerProps {
+  slideNumber: number;
+  setSlideNumber: (slideNumber: number) => void;
+}
+
+function SliderContainer(props: SliderContainerProps) {
+  const { slideNumber, setSlideNumber } = props;
+
+  const firstSlideRef = useSlideObserver(setSlideNumber, 1);
+  const secondSlideRef = useSlideObserver(setSlideNumber, 2);
+  const thirdSlideRef = useSlideObserver(setSlideNumber, 3);
+  const fourthSlideRef = useSlideObserver(setSlideNumber, 4);
+  const fifthSlideRef = useSlideObserver(setSlideNumber, 5);
+  const sixthSlideRef = useSlideObserver(setSlideNumber, 6);
+
+  useEffect(() => {
+    //slideNumber가 변경 될 때 마다 스크롤 옮겨주기
+    moveToElement(slideNumber);
+  }, [slideNumber]);
+
+  const moveToElement = (slideNumber: number) => {
+    const clickedSlideNum = slideNumber;
+    let clickedSlideRef: MutableRefObject<HTMLElement | null> | undefined;
+
+    switch (clickedSlideNum) {
+      case 1:
+        clickedSlideRef = firstSlideRef;
+        break;
+      case 2:
+        clickedSlideRef = secondSlideRef;
+        break;
+      case 3:
+        clickedSlideRef = thirdSlideRef;
+        break;
+      case 4:
+        clickedSlideRef = fourthSlideRef;
+        break;
+      case 5:
+        clickedSlideRef = fifthSlideRef;
+        break;
+      case 6:
+        clickedSlideRef = sixthSlideRef;
+        break;
+    }
+    {
+      clickedSlideRef &&
+        clickedSlideRef.current?.scrollIntoView({
+          block: 'start',
+        });
+    }
+  };
```

Comment:
이 부분 warning 뜨고 있었네용 아래처럼 바꿔주세요!

```suggestion
  useEffect(() => {
    const moveToElement = (slideNumber: number) => {
      const clickedSlideNum = slideNumber;
      let clickedSlideRef: MutableRefObject<HTMLElement | null> | undefined;

      switch (clickedSlideNum) {
        case 1:
          clickedSlideRef = firstSlideRef;
          break;
        case 2:
          clickedSlideRef = secondSlideRef;
          break;
        case 3:
          clickedSlideRef = thirdSlideRef;
          break;
        case 4:
          clickedSlideRef = fourthSlideRef;
          break;
        case 5:
          clickedSlideRef = fifthSlideRef;
          break;
        case 6:
          clickedSlideRef = sixthSlideRef;
          break;
      }

      clickedSlideRef &&
        clickedSlideRef.current?.scrollIntoView({
          block: 'start',
        });
    };

    moveToElement(slideNumber);
  }, [fifthSlideRef, firstSlideRef, fourthSlideRef, secondSlideRef, sixthSlideRef, slideNumber, thirdSlideRef]);
```

Reply:
1. React Hook useEffect has a missing dependency: 'moveToElement'. Either include it or remove the dependency array.
2. The 'moveToElement' function makes the dependencies of useEffect Hook (at line 34) change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'moveToElement' in its own useCallback() Hook.
3. React Hook useEffect has missing dependencies: 'fifthSlideRef', 'firstSlideRef', 'fourthSlideRef', 'secondSlideRef', 'sixthSlideRef', and 'thirdSlideRef'. Either include them or remove the dependency array.

Reply:
오 전 지금까지 페이지가 첫렌더링 될 때 동작한다고 생각하고 있었어요! useEffect 어렵네여